go/cmd/root.go

```diff
@@ -17,32 +17,77 @@ limitations under the License.
 package cmd
 
 import (
+	"fmt"
 	"os"
+	"time"
 
 	"github.com/spf13/cobra"
+
+	"github.com/vitessio/vt/go/web"
 )
 
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
 	// rootCmd represents the base command when called without any subcommands
+	var port int64
+	webserverStarted := false
+	ch := make(chan int, 2)
 	root := &cobra.Command{
 		Use:   "vt",
 		Short: "Utils tools for testing, running and benchmarking Vitess.",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			if port > 0 {
+				if webserverStarted {
+					return nil
+				}
+				webserverStarted = true
+				go startWebServer(port, ch)
+				<-ch
+			}
+			return nil
+		},
```

Comment:
I am not sure I understand why the web server is called in the root command (and thus for all other commands). We could instead do something similar to what we did for the graphs where we only spin up the web server at the end of summarize when the `*Summary` object is constructed. That way the logic would be local to summarize and the web server will be created at a time where we already have all the information we wish to render.